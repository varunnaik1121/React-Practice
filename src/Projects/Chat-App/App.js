import React, { useRef } from "react";
import { signup, db, useAuth, logout, login } from "./FB/Firebase";
import Profile from "./comps/Profile";
const App = () => {
  console.log(db);
  const emailRef = useRef(null);
  const passRef = useRef(null);
  const currentUser = useAuth();
  console.log(currentUser);
  const handleClick = async () => {
    try {
      await signup(emailRef.current.value, passRef.current.value);
    } catch {
      alert("cant signin again");
    }
  };

  const handleLogOut = async () => {
    try {
      await logout();
      emailRef.current.value = "";
      passRef.current.value = "";
    } catch (e) {
      console.log(e);
    }
  };
  const handleLogIn = async () => {
    try {
      await login(emailRef.current.value, passRef.current.value);
      emailRef.current.value = "";
      passRef.current.value = "";
    } catch (e) {
      console.log(e);
    }
  };

  return (
    <div>
      <h3>your email is {currentUser?.email}</h3>
      <input type="email" ref={emailRef}></input>
      <input type="password" ref={passRef}></input>
      <button onClick={handleClick}>signup</button>
      <button onClick={handleLogOut}>LogOut</button>
      <button onClick={handleLogIn}>LogIn</button>
      {currentUser && <Profile></Profile>}
    </div>
    
    
  );
};

export default App;
