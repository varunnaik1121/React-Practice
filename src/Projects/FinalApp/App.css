.post-wrapper{
    width: 85%;
    height: 100%;
    padding: 20px;
    /* border: 1px solid red; */
    margin: 0 auto;
    margin-top: 4%;background-color: white;
    box-shadow: 4px 4px 10px rgba(0,0,0,.15);
   
}

.title{
    font-size: 12px;
    text-transform: lowercase;

}
.post-wrapper{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}
.post-details{
    /* border: 1px solid red; */
    width: 100%;
    height: 100%;
  
    padding: 10px;
  
}
.title-wrapper{
    max-width: 14%;
    min-width: 14%;
    /* border: 1px solid red; */
    

}
.title-wrapper p{
    max-width: 100%;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    font-size: 12px;
}
.heading{
    font-size: 22px;
    text-transform: capitalize;
    margin: 20px 0;
    letter-spacing: 1px;
}
.headers{
    display: flex;
    justify-content:flex-start;
    width: 100%;
}
.headers span{
    width: 20%;
    /* border: 1px solid red; */
    text-align: center;
    padding: 10px 5px;
    text-transform: capitalize;
    background-color: #8f44fd;
    box-shadow: 3px 3px 5px rgba(0,0,0,.15);
    color: white;
    border-right: 2px solid white;
   
    
}
.title-wrapper{
    max-width: 25%;
    overflow: hidden;
}
p{
    margin-top: 15px;
}
.data{
    /* min-width: 200px; */
    /* border: 1px solid red; */
    text-align: left;
    max-width: 20%;
    min-width: 20%;
    font-size: 14px;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    margin-top: 20px;
    height: 60px;
    padding: 8px 6px 0 12px ;
    font-weight: 500 !important;
    
}
.data-wrapper{
    width: 100%;
    display: flex;
    justify-content: flex-start;
    overflow: hidden;
}
.data-wrapper span{
    width: 20%;
}
.btn{
    
   padding: 6px 16px;
   text-transform: capitalize;
   border: none;
   outline: none;
    text-align: center;
   margin: 20px auto 0 auto;
   border-radius: 2px;
   background-color: #8f44fd;
   box-shadow: 3px 3px 5px rgba(0,0,0,.15);
   color: white;
   cursor: pointer;
   margin-bottom: 50px;
}

.raw-wrapper{
    display: flex;
    flex-direction: column;
    
    width: 100vw;
    min-height: 100vh;
    overflow-x: hidden;
}
.raw{
    margin: 30px 0;
    padding: 10px 0px 20px 50px;
}
h4{
    margin:  0 auto;
    font-size: 24px;
}
.author{
    text-align: center !important;
  
}

.pagination{
  margin: 60px 0 100px 20%;
    
}
.btn-wrapper{
   margin: 30px auto;
}
.data:first-child{
    font-weight: 600;
}
