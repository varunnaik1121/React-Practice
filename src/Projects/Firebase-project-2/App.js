import React from 'react';
import { useState,useEffect } from 'react';
import UploadForm from './comps/uploadForm';
import ImageGrid from './comps/ImageGrid';

const App = () => {

  return <div>
        <UploadForm></UploadForm>
        <ImageGrid></ImageGrid>
  </div>;
};

export default App;
