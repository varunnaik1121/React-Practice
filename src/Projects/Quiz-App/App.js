import React,{useState,useEffect} from 'react'
import { BrowserRouter,Routes,Route } from 'react-router-dom'
import Home from './components/Home'
import { data } from './components/quetions'
import Quiz from './components/Quiz'


const App = () => {
   const [quetions,setQuetions] = useState(data);
 
  const [details,setDetails] = useState({
      name:'',
      language:''
  })
  const [finalQn,setFinalQn] = useState([])

  const getQuetionsByChoice = (language)=>{
   const particlarLangQuetion=quetions.filter((item)=>item.language===language)
   setFinalQn(particlarLangQuetion)
   localStorage.setItem('data',JSON.stringify(finalQn))
 

}
useEffect(()=>{
    getQuetionsByChoice(details.language);
   
    
   
},[details])
 
 




    return <>
        <BrowserRouter>
            <Routes>
                <Route exact path='/' element={<Home details={details} setDetails={setDetails}></Home>}></Route>
                <Route path='/quiz' element={<Quiz finalQn={finalQn} details={details}></Quiz>}></Route>
            </Routes>
        </BrowserRouter>
    </>
}

export default App
