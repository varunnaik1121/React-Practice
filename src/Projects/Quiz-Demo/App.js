import React from 'react'
import { BrowserRouter,Routes,Route } from 'react-router-dom'
import MainPage from './MainPage'
import QuizPage from './components/QuizPage'

const App = () => {
    return <>
        <BrowserRouter>
            <Routes>
                <Route exact path='/' element={<MainPage></MainPage>}></Route>
                <Route exact path='/quizPage' element={<QuizPage></QuizPage>}></Route>
            </Routes>
        </BrowserRouter>

    </>
}

export default App
