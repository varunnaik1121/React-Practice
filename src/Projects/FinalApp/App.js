import React from 'react'
import {useState,useEffect} from 'react';
import { Pagination } from '@mui/material';
import RawJson from './components/RawJson';
import Header from './components/header';
import './App.css'

const API = `https://hn.algolia.com/api/v1/search?query=bar&page=`;

const App = () => {

  const [show,setShow] = useState(false)
 
   const [data,setData] = useState([])
   const [currentIndex,setCurrentIndex] = useState(0)
   let tempArr = []
   let index=-1;
   const [pageCount,setPageCount] = useState(1)
   const [rawData,setrawData] = useState({})
   
    const fetchData = async ()=>{
      index=index+1;
      if(index>=50){
        return
      }
      try{
        const res = await fetch(API+index)
        const data = await res.json()
        tempArr=[...tempArr,data?.hits]
        setData(tempArr)
        setPageCount(pageCount+index)
      }
     catch(e){
       console.log(e)
     }
      

    }

        useEffect(() => {

            fetchData()
          
            const interval = setInterval(fetchData, 10000)
            
            return () => clearInterval(interval)
          }, []);
     
    const handlePageChange = (e,value)=>{
      console.log('hello')
      setCurrentIndex(value-1)
      
    }
    console.log( data?.[currentIndex])

    const handleClick = (e,index)=>{
      setShow(true)
      setrawData(JSON.stringify(data?.[currentIndex][index],null,4))
     
    }

    

    return <>
    

      {show ? <RawJson setShow={setShow} rawData={rawData}></RawJson>:data?.[currentIndex]&&<div>
         <div className='post-wrapper'>
          
           <Header></Header>
           <div className='post-details'>
           {
           data?.[currentIndex]?.map((item,index)=>{
             const {url,created_at,title,author} = item
             
             return <>
                  <div className='data-wrapper'>
                    <span className='data'>{index+1}. {title}</span>
                    <a className='data' href={url} target='_blank'>{url}</a>
                    <span className='data author'>{author}</span>
                    <span className='data'>{created_at}</span>
                    <button className='btn' onClick={(e)=>{handleClick(e,index)}} data-testid='select-btn'>select</button>
                  </div>
             </>
           })
         }
           </div>
         
         </div >
           {
             data?.[currentIndex]?<div className='pagination'>
             <Pagination
            count={ pageCount|| 0}
            color="secondary"
            onChange={handlePageChange}
            
          />
             </div>:null
           }

        </div>
  
}
        
      </>
}

export default App
