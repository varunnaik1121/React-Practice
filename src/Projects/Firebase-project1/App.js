import React from "react";
import Title from "./components/Title";
import UploadForm from "./components/UploadForm";

import { storage } from "./Firebase/config";
import "./App.css";
const App = () => {
  console.log(storage);

  return (
    <>
      <div>hello from firebase App</div>
      <Title></Title>
      <UploadForm></UploadForm>
    </>
  );
};

export default App;
