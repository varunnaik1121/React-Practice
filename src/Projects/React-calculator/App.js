import React from "react";
import Calculator from "./comps/Calculator";
import Display from "./comps/Display";
import ButtonWrapper from "./comps/ButtonWrapper";
import "./comps/App.css";
import { useGlobal } from "./comps/context";
import Button from "./comps/Button";
const App = () => {
  const symbols = [
    [
      ["C", "+-", "%", "/"],
      [7, 8, 9, "x"],
      [4, 5, 6, "-"],
      [1, 2, 3, "+"],
      [0, ".", "="],
    ],
  ];
  const { handleClick, calc, inputRef } = useGlobal();
  return (
    <>
      <Calculator>
        <Display
          value={
            calc.num
              ? Number(calc.num).toLocaleString()
              : Number(calc.result).toLocaleString()
          }
          inputRef={inputRef}
        ></Display>
        <ButtonWrapper>
          {symbols.flat(Infinity).map((value, index) => {
            return (
              <Button
                key={index}
                className={value === "=" ? "equals" : ""}
                innerValue={value}
                handleClick={handleClick}
              ></Button>
            );
          })}
        </ButtonWrapper>
      </Calculator>
    </>
  );
};

export default App;
