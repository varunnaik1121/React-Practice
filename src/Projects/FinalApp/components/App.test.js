import '@testing-library/jest-dom/extend-expect';
 import React from "react";
import { cleanup, getByTestId, render, screen } from "@testing-library/react";
import Header from './header'
import Json from './RawJson'
import App from '../App'
 
const rawData = {
    "title": "freecode",
    "url": "www.freecodecamp.com",
    "author": "jhon",
    "created_at": "11-06-2011",
  };

test('Test to render App',()=>{
            render(<App></App>)
           screen.debug()
         
})


  
describe('testing header component ',()=>{
    render(<Header></Header>)
    expect(screen.getByText(/list data/i)).toBeInTheDocument()
    expect(screen.getByText(/list data/i)).toHaveClass('heading')
    expect(screen.getByText(/list data/i)).toHaveAttribute('class')
    expect(screen.getByText(/title/i)).toHaveTextContent('title')
})

describe('testing rawJSon without any props ',()=>{
    test('rawJson',()=>{
        render(<Json ></Json>)
       
        expect(screen.getByTestId('main-button')).toHaveAttribute('class');
       
    
      
    })
     
  })
  

describe('testing rawJSon with props ',()=>{
  test('rawJson',()=>{
      render(<Json {...rawData} setShow={true}></Json>)
     
      expect(screen.getByTestId('main-button')).toHaveClass('btn');
      expect(screen.getByTestId('main-button')).toHaveTextContent('back to main page')
     
  
    
  })
   
})


describe('snapshot testing ',()=>{

    test('rawJson',()=>{
        const {container} =render(
           <Json {...rawData} setShow={true}></Json>
        );
        expect(container).toMatchSnapshot()
       
        
       
    
      
    })
     
  })
  
  