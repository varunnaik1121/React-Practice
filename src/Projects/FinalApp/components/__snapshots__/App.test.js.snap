// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing  rawJson 1`] = `
<div>
  <div
    class="raw-wrapper"
  >
    <div
      class="btn-wrapper"
    >
      <button
        class="btn"
        data-testid="main-button"
      >
        back to main page
      </button>
    </div>
    <h4>
      Raw JSON for the selected post......
    </h4>
    <pre
      class="raw"
    >
      <code />
    </pre>
  </div>
</div>
`;
